includes/preprocessor_base.inc

```diff
@@ -107,20 +107,61 @@ abstract class IslandoraBatchPreprocessor {
   /**
    * Get an identifier we can use for an object.
    *
+   * @param AbstractFedoraObject $object
+   *   The object to be later processed and added to Fedora.
+   *
    * @return string
    *   An id to use in the back-end (Fedora).
    */
-  protected function getId() {
-    // Not specifying a namespace should result in one from the default
-    // declared in the fedora.fcfg ("changeme" by default).
-    $namespace = (isset($this->parameters['namespace'])) ?
-      $this->parameters['namespace'] :
-      NULL;
+  protected function getId(AbstractFedoraObject $object) {
+    // Command line argument overrides Collection Policy.
+    if (isset($this->parameters['namespace'])) {
+      $namespace = $this->parameters['namespace'];
+    }
+    else {
+      $namespace = $this->getNamespace($object);
+    }
 
     // TODO: Implement some form of caching, so we can get multiple at a time.
     return $this->connection->repository->api->m->getNextPid($namespace);
   }
 
+
+  /**
+   * Try to get the namespace from the collection policy.
+   *
+   * @param AbstractFedoraObject $object
+   *   The object to be later processed and added to Fedora.
+   *
+   * @return string|NULL
+   *   The namespace with which the object should be ingested.
+   *   NULL if we use the Fedora configured default.
+   */
+  protected function getNamespace(AbstractFedoraObject $object) {
+    // Get namespace based on collection.
+    $collection_policies =& drupal_static(__METHOD__, array());
+    $parent_pid = $this->parameters['parent'];
```

Comment:
Quick note @adam-vessey: $this->parameters['parent'] defaults to the root collection anyway
https://github.com/Islandora/islandora_batch/blob/36d41bb9436ab3670528e141e990abe4c5f259b5/islandora_batch.drush.inc#L118 ?